src/transformers/integrations/bitsandbytes.py

```diff
@@ -74,13 +76,14 @@ class `Int8Params` from `bitsandbytes`.
                 new_value = old_value.to(device)
             elif isinstance(value, torch.Tensor):
                 new_value = value.to("cpu")
-                if value.dtype == torch.int8:
+                if value.dtype in (torch.int8, torch.uint8):
                     is_8bit_serializable = version.parse(importlib.metadata.version("bitsandbytes")) > version.parse(
                         "0.37.2"
                     )
```

Comment:
I think we should check for the correct bnb version when it comes to 4bit as well, if the PR https://github.com/TimDettmers/bitsandbytes/pull/753 gets merged, it should probably be greater than the current `0.41.1` version

Reply:
added checks with `is_4bit_serializable `